### ISSUE TITLE
Traversal over a segment of the sphere.

### ISSUE THREAD

**Issue by cgyurgyik** [spherical-volume-rendering/svr-algorithm#102]
One thing we talked about in our meeting this week was only traversing through a "slice" of the sphere. We need to discuss how this can be implemented, and what needs to change in our current algorithm.

**Comment by ak-2485** [comment]
![File_001 (1)](https://user-images.githubusercontent.com/25615362/79505367-44c0d700-8002-11ea-9b03-9dee370b5f21.png)

**Comment by ak-2485** [comment]
@cgyurgyik @nukenukenukelol We had the question of how t_end would be determined @nukenukenukelol was correct in that we already have prescribed bounds for theta, so it would exit after that bound is hit; t_end can be some dummy variable.

**Comment by ak-2485** [comment]
@matthewturk how/where are AMR calls handled for walk_volume in yt?

**Comment by matthewturk** [comment]
@ak-2485 So `walk_volume` only ever gets uniform grids -- we do a decomposition at an earlier phase that breaks AMR up into non-overlapping grids we call "blocks" or "bricks."  So from the `walk_volume` perspective, AMR doesn't enter into it.

**Comment by ak-2485** [comment]
The blocks are non-uniform?

**Comment by matthewturk** [comment]
Sorry, let me rephrase.  When `walk_volume` is called, it is given a single block that is uniform in dx, dy, dz.  But, we call `walk_volume` multiple times, and when we call it the first time, that block may be different than the next time we call it.

**Comment by cgyurgyik** [comment]
From my understanding, this approach will replace ```min_bound, max_bound```, which are currently unused. We will define a new type of bound, maybe like this:

```
    // Represents a bound for the sphere using voxel identifications.
    struct VoxelBound {
        std::size_t radial_voxel;
        std::size_t angular_voxel;
        std::size_t azimuthal_voxel;
    };
```

The algorithm will pass in ```VoxelBound min_bound, VoxelBound max_bound```, and use those
to determine how far we want the ray to intersect.

If the ray were to intersect the entire sphere,
```
VoxelBound min_bound = {1, 0, 0};
VoxelBound max_bound = {1, num_angular_sections - 1, num_azimuthal_sections - 1};
```
We could provide these as default values to our algorithm, but I'm personally not a fan of defaulted values.

If the ray wants to intersect say a sector of the sphere,
Let num_radial_sections = num_angular_sections = num_azimuthal_sections = 4. Then,
```
VoxelBound min_bound = {1, 0, 0};
VoxelBound max_bound = {4, 1, 1};
```
Rays will intersect the sphere in the first quadrant, as pictured below:
![image](https://user-images.githubusercontent.com/37983775/81315439-4ce7c180-9058-11ea-8b6a-2d64f32b78d5.png)

Our algorithm will then end early in two areas:
1. During initialization phase,
```
for each voxel_ID:
      if voxel_ID < min_bound, return.
```
2. During the traversal phase,
```
for each voxel_ID:
     if voxel_ID >= max_bound, return.
```

**Comment by matthewturk** [comment]
This makes sense to me.  I would note that we can convert pretty easily
from one to the other, from an r_min to a voxel bound, and that's something
we could do in the higher-level code.

**Comment by cgyurgyik** [comment]
> This makes sense to me. I would note that we can convert pretty easily from one to the other, from an r_min to a voxel bound, and that's something we could do in the higher-level code.

Yeah that's a good point. If a user provides numbers, say, [0, 2pi], we can convert that to the corresponding voxels as well in an intermediary function.

**Comment by cgyurgyik** [comment]
I had a question about assumptions we can make @matthewturk . 
1. Will the client use directions from say, (pi/4, -pi/4) -> (pi/4, pi/4) ? In other words, do we need to cover the case where the boundaries include the transition from voxel N-1 to voxel 0?

**Comment by matthewturk** [comment]
We can ensure that those transitions don't occur.
